api/src/modules/h3-data/h3-data.repository.ts

```diff
@@ -285,49 +303,60 @@ export class H3DataRepository extends Repository<H3Data> {
-    //   .createQueryBuilder()
-    //   .select(`sum(h3table.${h3ColumnName})`, 'sum')
-    //   .from(h3TableName.toLowerCase(), 'h3table')
-    //   .innerJoin(
-    //     (subQuery: SelectQueryBuilder<any>) => {
-    //       return subQuery
-    //         .select('h3_uncompact(gr."h3Compact"::h3index[], 6)', 'h3index')
-    //         .from(GeoRegion, 'gr')
-    //         .where('gr.id = :geoRegionId');
-    //     },
-    //     'georegionh3',
-    //     'h3table.h3index = georegionh3.h3index',
-    //   )
-    //   .setParameter('geoRegionId', geoRegionId)
-    //   .getRawOne();
-    // TODO: Refactor this using the query builder
-    const query: string = `
-      SELECT locations.tonnage * sum(materials.bio_risk_PDF_t) AS value
-      FROM (SELECT sr.id, sr.tonnage, unnest(gr."h3Compact"::h3index[]) h3index
-            FROM sourcing_records sr
-                   LEFT JOIN sourcing_location sl ON sl.id = sr."sourcingLocationId"
-                   LEFT JOIN geo_region gr ON gr.id = sl."geoRegionId"
-            WHERE sr.id = '${sourcingRecordId}') locations
-             JOIN
-           (SELECT ha.h3index,
-                   ind.${indicatorH3Table.h3columnName} *
-                   ((def.hansen_loss_2019 * ha.${harvestH3Table.h3columnName}) / (sum(prod.${producerH3Table.h3columnName}) over ())) bio_risk_PDF_t
-            FROM ${harvestH3Table.h3tableName} ha
-                   LEFT JOIN ${producerH3Table.h3tableName} prod
-                             ON prod.h3index = ha.h3index
-                   LEFT JOIN h3_grid_deforestation_global def
-                             on def.h3index = ha.h3index --replace by deforestation table
-                   LEFT JOIN ${indicatorH3Table.h3tableName} ind ON ind.h3index = ha.h3index
-            WHERE prod.${producerH3Table.h3columnName} > 0
-              AND ha.${harvestH3Table.h3columnName} > 0
-              AND def.hansen_loss_2019 > 0
-              AND ind.${indicatorH3Table.h3columnName} > 0) materials --replace by ind column name
-           ON materials.h3index = locations.h3index
-      GROUP BY locations.id, locations.tonnage
-    `;
-
-    const result: { value: number }[] = await getManager().query(query);
+    const result: { value: number }[] = await getManager()
+      .createQueryBuilder()
+      .select('locations.tonnage * sum(materials.bio_risk_PDF_t)', 'value')
+      .from((subQuery: SelectQueryBuilder<any>) => {
+        return subQuery
+          .select('sr.id', 'id')
+          .addSelect('sr.tonnage', 'tonnage')
+          .addSelect('unnest(gr."h3Flat"::h3index[])', 'h3index')
+          .from('sourcing_records', 'sr')
+          .leftJoin(
+            'sourcing_location',
+            'sl',
+            'sl.id = sr."sourcingLocationId"',
+          )
+          .leftJoin('geo_region', 'gr', 'gr.id = sl."geoRegionId"')
+          .where(`sr.id = :sourcingRecordId`, {
+            sourcingRecordId: sourcingRecordId,
```

Comment:
could be this be shorthanded to 
```suggestion
{ sourcingRecordId }
```